src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/resthandler/RestStopTransformActionIT.kt

```diff
@@ -0,0 +1,208 @@
+            assertEquals("Transform should have stayed finished", TransformMetadata.Status.FINISHED, metadata.status)
+        }
+    }
+
+    @Throws(Exception::class)
+    fun `test stopping a failed transform`() {
+        // Create a transform that will fail because no source index
+        val transform = randomTransform().copy(
+            id = "test_stopping_a_failed_transform",
+            jobSchedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES),
+            enabled = true,
+            enabledAt = Instant.now(),
+            metadataId = null
+        ).let { createTransform(it, it.id) }
+        updateTransformStartTime(transform)
+
+        // Assert it's in failed
+        waitFor {
+            val updatedTransform = getTransform(transform.id)
+            val metadata = getTransformMetadata(updatedTransform.metadataId!!)
+            assertEquals("Transform never failed", TransformMetadata.Status.FAILED, metadata.status)
+        }
+
+        // Stop transform
+        val response = client().makeRequest("POST", "$TRANSFORM_BASE_URI/${transform.id}/_stop")
+        assertEquals("Stop transform failed", RestStatus.OK, response.restStatus())
+        val expectedResponse = mapOf("acknowledged" to true)
+        assertEquals(expectedResponse, response.asMap())
+
+        // Assert transform still failed status
+        waitFor {
+            val updatedTransform = getTransform(transform.id)
+            val metadata = getTransformMetadata(updatedTransform.metadataId!!)
+            assertEquals("Transform should have stayed failed", TransformMetadata.Status.FAILED, metadata.status)
+        }
+    }
+
+    // ISSUE: Goes straight to failed before test can pick up STARTED status
+
+    @Throws(Exception::class)
+    fun `test stopping transform with metadata`() {
+        generateNYCTaxiData("source")
+        val transform = randomTransform().copy(
+            id = "basic_term_query",
+            schemaVersion = 1L,
+            enabled = true,
+            jobSchedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES),
+            updatedAt = Instant.now(),
+            enabledAt = Instant.now(),
+            description = "basic search test",
+            sourceIndex = "source",
+            targetIndex = "target",
+            metadataId = null,
+            roles = emptyList(),
+            pageSize = 10,
+            groups = listOf(
+                Terms(sourceField = "store_and_fwd_flag", targetField = "flag")
+            )
+        ).let { createTransform(it, it.id) }
+
+        updateTransformStartTime(transform)
+
+        waitFor {
+            val transformJob = getTransform(transformId = transform.id)
+            assertNotNull("Transform job doesn't have metadata set", transformJob.metadataId)
+            val transformMetadata = getTransformMetadata(transformJob.metadataId!!)
+            assertEquals("Transform is not STARTED", TransformMetadata.Status.STARTED, transformMetadata.status)
```

Comment:
I am seeing these errors in the workflow ``java.lang.AssertionError: Transform is not STARTED expected:<started> but was:<finished>``, probably job is completed by the time waitFor block is able to find the metadata associated with the job. 

Can you try to change the transform job so it runs longer?
```
generateNYCTaxiData("source_test_stopping_running_transform")
val transform = randomTransform().copy(
            id = "test_stopping_running_transform",
            schemaVersion = 1L,
            enabled = true,
            jobSchedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES),
            updatedAt = Instant.now(),
            enabledAt = Instant.now(),
            description = "basic search test",
            sourceIndex = "source"_test_stopping_running_transform",
            targetIndex = "target_test_stopping_running_transform",
            metadataId = null,
            roles = emptyList(),
            pageSize = 1,
            groups = listOf(
                Terms(sourceField = "tpep_pickup_datetime", targetField = "pickup time", period="60m"),
                Terms(sourceField = "store_and_fwd_flag", targetField = "flag")
            )
        ).let { createTransform(it, it.id) }
```